src/syslog.rs

```diff
@@ -0,0 +1,178 @@
+) -> Result<()> {
+    let ident = ident.map_or(ptr::null(), |ident| ident.as_ptr());
+    let logopt = logopt.bits();
+    let facility = facility as libc::c_int;
+    unsafe {
+        libc::openlog(ident, logopt, facility);
+    }
+    Ok(())
+}
+
+/// Writes message to the system message logger.
+///
+/// The message is then written to the system console, log files, logged-in users, or forwarded
+/// to other machines as appropriate.
+///
+/// # Examples
+///
+/// ```rust
+/// use nix::syslog::{openlog, syslog, Facility, LogFlags, Severity};
+///
+/// let flags = LogFlags::LOG_PID;
+/// openlog(None, flags, Facility::LOG_USER).unwrap();
+/// syslog(Severity::LOG_EMERG, "Hello, nix!").unwrap();
+///
+/// // use `format!` to format the message
+/// let name = "syslog";
+/// syslog(Severity::LOG_EMERG, &format!("Hello, {name}!")).unwrap();
+/// ```
+pub fn syslog<P, S>(priority: P, message: &S) -> Result<()>
+where
+    P: Into<Priority>,
+    S: AsRef<OsStr> + ?Sized,
+{
+    let priority = priority.into();
+    let formatter = OsStr::new("%s");
+    let message = OsStr::new(message);
+    formatter.with_nix_path(|formatter| {
+        message.with_nix_path(|message| unsafe {
+            libc::syslog(priority.0, formatter.as_ptr(), message.as_ptr())
+        })
+    })??;
+    Ok(())
+}
+
+/// Closes the log file.
+pub fn closelog() {
+    unsafe { libc::closelog() }
+}
+
+/// The priority for a log message.
+#[derive(Debug, Clone, Copy)]
+pub struct Priority(libc::c_int);
+
+impl Priority {
+    /// Create a new priority from a facility and severity level.
+    pub fn new(severity: Severity, facility: Facility) -> Self {
+        let priority = (facility as libc::c_int) | (severity as libc::c_int);
+        Priority(priority)
+    }
+}
+
+impl From<Severity> for Priority {
+    fn from(severity: Severity) -> Self {
+        let priority = severity as libc::c_int;
+        Priority(priority)
+    }
+}
+
+libc_bitflags! {
+    /// Options for system logging.
+    pub struct LogFlags: libc::c_int {
+        /// Log the process id with each message: useful for identifying instantiations of
+        /// daemons.
+        LOG_PID;
+        /// If syslog() cannot pass the message to syslogd(8) it will attempt to write the
+        /// message to the console ("/dev/console").
+        LOG_CONS;
+        /// The converse of [`LOG_NDELAY`][LogFlags::LOG_NDELAY]; opening of the connection is
+        /// delayed until `syslog` is called.
+        ///
+        /// This is the default, and need not be specified.
+        LOG_ODELAY;
+        /// Open the connection to syslogd(8) immediately. Normally the open is delayed until
+        /// the first message is logged. Useful for programs that need to manage the order in
+        /// which file descriptors are allocated.
+        LOG_NDELAY;
+        /// Write the message to standard error output as well to the system log.
+        #[cfg(all(not(target_os = "redox"), not(target_os = "illumos")))]
```

Comment:
Let's do:

```suggestion
        #[cfg(not(any(target_os = "redox", target_os = "illumos")))]
```

This is easier to read and feels more natural.